uaclient/cli.py

```diff
@@ -103,26 +101,62 @@ def __init__(
         )
 
         self.base_desc = base_desc
-        self.non_beta_services_desc = non_beta_services_desc
-        self.beta_services_desc = beta_services_desc
 
     def error(self, message):
         self.print_usage(sys.stderr)
         self.exit(2, message + "\n")
 
     def print_help(self, file=None, show_all=False):
-        desc_vars = [
-            self.base_desc,
-            self.non_beta_services_desc,
-            self.beta_services_desc,
-        ]
-        if any(desc_vars):
-            services = sorted(self.non_beta_services_desc)
+        if self.base_desc:
```

Comment:
Will this value ever be `None` ? It seems we get the `__doc__` attribute of cli module whenever we create the `UAArgumentParser`.

Reply:
It will be None when the parser is called implicitly.
I have found that when calling `ua status --help` for instance, it might get `None`'d by argparse somehow.
I kept the same logic @blackboxsw had there before (looking for any of the desc_vars), just adapting it because what we really want to know is if `base_desc` has been set - the services will only be used in that case anyway.

Reply:
I think what is happening is that only the main parser will have the `base_desc` value. All subparsers won't. When the `parser.parse_args` is called, we end up calling `print_help` on the subparser, which doesn't have the `base_desc`  value populated.

Which seems to work fine for what we are trying to achieve

Reply:
Makes sense!